test/general/sycl_iterator/sycl_iterator_for.pass.cpp

```diff
@@ -982,7 +982,7 @@ DEFINE_TEST(test_adjacent_difference)
         Iterator1ValueType fill_value{1};
         Iterator2ValueType blank_value{0};
 
-        auto __f = [](Iterator1ValueType& a, Iterator1ValueType& b) -> Iterator2ValueType { return a + b; };
+        auto __f = [](const Iterator1ValueType& a, const Iterator1ValueType& b) -> Iterator2ValueType { return a + b; };
```

Comment:
or we can write shorter
`auto __f = [](auto& a, auto& b)`

Reply:
I suppose the idea is to have a test which imposes the requirements for an algorithm as described in #2184. We need that "const" to test a more restrictive case.

Reply:
I guess Mikhail rather suggested writing `auto` instead of `Iterator1ValueType` - I am indifferent here.

Reply:
I meant "auto" instead of "const Iterator1ValueType"...
Because "auto& a" wil be deduced into "const int& a" in case of passing "const int", for example.

But I don't insist on.